### ISSUE TITLE
`score()` weights who made the video above what it is about — an allowlisted channel beats a perfect topic match

### ISSUE THREAD

**Issue by opendroid** [opendroid/the-infinity#432]
Six picks across four batches have now had to be rescued by hand from below the scorer's top choice. That is not curation noise. **It is arithmetic, and it is provable from the constants.**

## The weights, as written

`web/scripts/find-video-explainers.mjs`, `score()`:

| term | value |
|---|---|
| allowlisted channel | **+0.40** |
| named author | +0.25 |
| **topic overlap** | **0.3 × overlap → max +0.30** |
| teachable length | +0.20 |
| views prior | `min(0.1, log10(views)/100)` → **~0.07 in practice** |
| almost unwatched (<500 views) | −0.25, *not applied to trusted channels* |

**Being on the allowlist is worth more than being about the right subject.** 0.40 > 0.30.

## What that produces, exactly

An allowlisted video of teachable length with **zero** topic overlap:

```
0.40 (allowlist) + 0.00 (topic) + 0.20 (length) + 0.073 (views) = 0.673
```

A video from any other channel that matches the topic **perfectly**:

```
0.00 + 0.30 (topic) + 0.20 (length) + 0.073 = 0.573
```

**0.67 beats 0.57.** A 3Blue1Brown video about something else outranks an exact match from anyone not on the list, and no amount of relevance closes the gap — the hard ceiling for an untrusted candidate is 0.60, below the 0.70 an allowlisted one reaches while matching nothing at all.

Confirmed against the live run rather than derived on paper. `early-exit` → 3Blue1Brown's *But what is a neural network?* scored **0.67** on **0% overlap**, which is `0.40 + 0.20 + log10(24,210,799)/100` to three decimals.

## Measured across the 518-candidate sweep

| | |
|---|---|
| targets with candidates | 181 |
| **top pick had ZERO topic overlap** | **8** |
| **a strictly better topic match was ranked lower** | **14** |

Every one of the eight zero-overlap winners is 3Blue1Brown or StatQuest:

```
early-exit                   0.67  But what is a neural network?          — 3Blue1Brown
emergent-abilities           0.67  Large Language Models explained briefly — 3Blue1Brown
bayesian-inference           0.67  Bayes theorem, the geometry of …        — 3Blue1Brown
convolutional-network        0.67  But what is a convolution?             — 3Blue1Brown
diffusion-model              0.66  But how do AI images and videos …      — 3Blue1Brown
logit                        0.66  StatQuest: Logistic Regression         — StatQuest
image-generation-evaluation  0.66  ML Fundamentals: The Confusion Matrix  — StatQuest
dp-sgd                       0.66  Stochastic Gradient Descent …          — StatQuest
```

And the inversions, with the better match the scorer buried:

| concept | won | lost |
|---|---|---|
| `bayesian-inference` | 0.67 **0%** 3Blue1Brown — *Bayes theorem* | 0.55 **100%** Steve Brunton — *Bayesian Inference: Overview* |
| `dp-sgd` | 0.66 **0%** StatQuest — *Stochastic Gradient Descent* | 0.53 **100%** Google TechTalks — *DP-SGD Privacy Analysis is Tight!* |
| `hierarchical-rl` | 0.81 **50%** StatQuest — *Hierarchical **Clustering*** | 0.53 **100%** chandar-lab — *Hierarchical RL, Lecture 11* |
| `batch-normalization` | 0.80 **50%** DeepLearningAI — *Why Does Batch Norm Work?* | 0.55 **100%** deeplizard — *Batch Normalization explained* |
| `convolutional-network` | 0.67 **0%** 3Blue1Brown — *But what is a convolution?* | 0.56 **100%** codebasics — *Simple explanation of CNNs* |

These are good videos. They are just not about the concept they were offered for, and the scorer cannot tell the difference because it is not weighted to.

## Why the allowlist was worth 0.40, and why that reasoning does not survive

The bonus is doing a real job: it is a proxy for *this will actually teach you*, which no other term captures, and it correctly promotes 3Blue1Brown's convolution video **for `convolution`**. The mistake is letting a proxy for quality outweigh the question of subject. Quality only matters once relevance is established — a superb explanation of a different topic is worth zero to a reader on this page, and ADR-0017 is explicit that `explainers` is *where to go and be taught **it***.

## What to change

1. **Make relevance a gate, not a term.** Zero topic overlap disqualifies a candidate outright, whatever the channel. That alone removes all eight zero-overlap winners and costs nothing correct.
2. **Re-balance so relevance dominates.** Overlap to `0.5 ×` and the allowlist to `+0.25`, so a perfect match from an unknown channel (0.50) beats a half-match from an allowlisted one (0.25 + 0.25 = 0.50 — tie) and beats a quarter-match outright. The exact numbers want tuning against the data below, not choosing here.
3. **Keep the unwatched floor.** #395's argument still holds; nothing in this issue disturbs it.

## This can be validated without spending any quota

**`video-candidates.concepts.json` already holds 518 scored candidates with their raw fields.** Re-scoring is offline — no API calls, no budget — so a proposed weighting can be run against the whole existing set and diffed against the current ranking before it is committed. The six hand-rescued picks are a labelled test set that already exists:

| concept | the right answer |
|---|---|
| `hierarchical-rl` | chandar-lab — *Hierarchical RL, Lecture 11* |
| `job-scheduling` | Microsoft Research — *Scheduling for Large-Scale ML Training* |
| `ablation` | *(from the earlier batch)* |
| `alibi` | *(from the earlier batch)* |
| `adam` | *(from the earlier batch)* |
| `active-learning` | *(from the earlier batch)* |

## Acceptance criteria

- [ ] A candidate with zero topic overlap cannot be the top pick for a target, on any channel
- [ ] For each of the inversions listed above, the better topic match ranks first under the new weighting
- [ ] The videos that are correct today stay correct — 3Blue1Brown's convolution video still wins `convolution`, StatQuest still wins `logistic-regression`
- [ ] Table-driven tests over the scoring constants, including a case that fails if the allowlist bonus is ever raised above the maximum overlap term again
- [ ] Planted: restoring `allowlist 0.40 / overlap 0.3` puts the zero-overlap winners back on top and fails the tests
- [ ] The re-scored ranking is diffed against the existing 518 candidates and the change in top picks is reported in the PR — no quota spent

Refs #384, #386, #395, #400, #403, #429, #430, ADR-0017.
